### PR TITLE
store date of nightly version in separate field

### PR DESCRIPTION
**Pull request by syphar**
this is helpful for two things I'm working on: 

1. automatic rebuilds for old releases. I want to configure a "max nightly date" to rebuild up to. 
2. a command to queue rebuilds for broken nightly versions, just via using the date.

### CODE REVIEW

File: src/db/add_package.rs [hunk 1]
```diff
@@ -244,6 +244,17 @@ pub(crate) async fn finish_build(
     debug!("updating build after finishing");
     let hostname = hostname::get()?;
 
+    let rustc_date = match parse_rustc_date(rustc_version) {
```
Comment (GuillaumeGomez) [inline src/db/add_package.rs:247]: Do we not want to return early in case the date couldn't be parsed instead?

Comment (syphar) [reply src/db/add_package.rs:247]: I wasn't sure about the possible edge cases. While I tested the parsing with all existing `rustc_version` contents in our production `builds` table, we have a handful of records where the version is empty. 

So for now I wanted to just log an error. In the end, the build finished, even when the version couldn't be parsed.

Comment (GuillaumeGomez) [reply src/db/add_package.rs:247]: Oh well. It's fine I suppose then. Please add a comment with this explanation in the code then so future us won't wonder what's going on. ;)

Comment (syphar) [reply src/db/add_package.rs:247]: added a comment

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**